src/lib/y2network/network_manager/connection_config_writers/wireless.rb

```diff
@@ -58,25 +47,84 @@ def write_eap_auth_settings(_conn)
+          auth_mode = conn.auth_mode || :open
+          meth = "write_#{auth_mode}_auth_settings".to_sym
           send(meth, conn) if respond_to?(meth, true)
         end
 
+        # Writes autentication settings for WPA-EAP networks
+        #
+        # @param conn [Y2Network::ConnectionConfig::Base] Configuration to write
+        def write_eap_auth_settings(conn)
+          # FIXME: incomplete
+          file.wifi_security["key-mgmt"] = "wpa-eap"
+          section = file.section_for("802-1x")
+
+          section["eap"] = conn.eap_mode.downcase if conn.eap_mode
+          section["phase2-auth"] = conn.eap_auth if conn.eap_auth
+          section["password"] = conn.wpa_password if conn.wpa_password && conn.eap_mode != "TLS"
+          section["anonymous-identity"] = conn.wpa_anonymous_identity if conn.eap_mode == "TTLS"
+          section["identity"] = conn.wpa_identity if conn.wpa_identity
+          section["ca-cert"] = conn.ca_cert if conn.ca_cert
+
+          return unless conn.eap_mode == "TLS"
+
+          section["client-cert"] = conn.client_cert
+          section["private-key"] = conn.client_key
+          section["private-key-password"] = conn.client_key_password
+        end
+
         # Writes autentication settings for WPA-PSK networks
         #
         # @param conn [Y2Network::ConnectionConfig::Base] Configuration to write
         def write_psk_auth_settings(conn)
-          file.wifi_security["auth-alg"] = "open"
           file.wifi_security["key-mgmt"] = "wpa-psk"
+          file.wifi_security["auth-alg"] = "open"
           file.wifi_security["psk"] = conn.wpa_psk
         end
 
         # Writes autentication settings for WEP networks
         #
-        # @param _conn [Y2Network::ConnectionConfig::Base] Configuration to write
-        def write_shared_auth_settings(_conn); end
+        # @param conn [Y2Network::ConnectionConfig::Base] Configuration to write
+        def write_wep_auth_settings(conn)
+          file.wifi_security["key-mgmt"] = "none"
+          default_key_idx = conn.default_key || 0
+          file.wifi_security["wep-tx-keyidx"] = default_key_idx.to_s if !default_key_idx.zero?
+          conn.keys.each_with_index do |v, i|
+            next if v.to_s.empty?
+
+            file.wifi_security["wep-key#{i}"] = v.gsub(/^[sh]:/, "")
+          end
+          passphrase_used = conn.keys[default_key_idx].to_s.start_with?(/h:/)
+          # see https://developer.gnome.org/libnm/stable/NMSettingWirelessSecurity.html#NMWepKeyType
+          # 1: Hex or ASCII, 2: 104/128-bit Passphrase
+          file.wifi_security["wep-key-type"] = passphrase_used ? "2" : "1"
+
+          true
+        end
+
+        # Writes autentication settings for WEP networks (open auth)
+        #
+        # @param conn [Y2Network::ConnectionConfig::Base] Configuration to write
+        def write_open_auth_settings(conn)
+          return if (conn.keys || []).compact.all?(&:empty?)
```

Comment:
You might be interested in adding a method to the connection class if needed in order to improve readability. Especially given that you use this construction in several places.

Reply:
Yep, I was already thinking on that, so, time to do it ;)